vendor/github.com/outbrain/golib/sqlutils/sqlutils.go

```diff
@@ -195,34 +196,33 @@ func ScanRowsToMaps(rows *sql.Rows, on_row func(RowMap) error) error {
 
 // QueryRowsMap is a convenience function allowing querying a result set while poviding a callback
 // function activated per read row.
-func QueryRowsMap(db *sql.DB, query string, on_row func(RowMap) error, args ...interface{}) error {
-	var err error
+func QueryRowsMap(db *sql.DB, query string, on_row func(RowMap) error, args ...interface{}) (err error) {
 	defer func() {
 		if derr := recover(); derr != nil {
 			err = errors.New(fmt.Sprintf("QueryRowsMap unexpected error: %+v", derr))
 		}
 	}()
 
-	rows, err := db.Query(query, args...)
+	var rows *sql.Rows
+	rows, err = db.Query(query, args...)
 	defer rows.Close()
 	if err != nil && err != sql.ErrNoRows {
 		return log.Errore(err)
 	}
 	err = ScanRowsToMaps(rows, on_row)
-	return err
+	return
```

Comment:
@zmoazeni I believe this func should `return err` or `return nil`. Can you confirm?

Reply:
@timvaillancourt That seems to be the case.